plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/files/XmlBeanDefinitionScannerAgent.java

```diff
@@ -126,48 +125,60 @@ public static void registerXmlBeanDefinitionScannerAgent(XmlBeanDefinitionReader
         }
     }
 
-    public static void reloadClass(String className) {
-        for (XmlBeanDefinitionScannerAgent agent : instances.values()) {
-            if (!agent.beansRegistered.containsValue(className)) {
+    public static Set<String> reloadXmls(DefaultListableBeanFactory beanFactory, Set<URL> urls, Set<String> resourcePaths) {
```

Comment:
This method signature is confusing. In my opinion, we should pass either a set of URLs or a set of String as the 2nd parameters but should avoid both. In this way, we could avoid dup code in its impl.

Reply:
I think we can simplify the code as below:

```java
    public static Set<String> reloadXmls(DefaultListableBeanFactory beanFactory, Set<String> paths) {
        Set<XmlBeanDefinitionScannerAgent> agents = new HashSet<>(paths.size());
        for (String path : paths) {

            XmlBeanDefinitionScannerAgent xmlBeanDefinitionScannerAgent = getAgent(beanFactory, path);
            if (xmlBeanDefinitionScannerAgent == null) {
                LOGGER.warning("url " + path + " is not associated with any XmlBeanDefinitionScannerAgent, not reloading");
                continue;
            }
            try {
                LOGGER.trace("Reloading XML {} since property file changed: {}", xmlBeanDefinitionScannerAgent.url, path);
                if (agents.add(xmlBeanDefinitionScannerAgent)) {
                    xmlBeanDefinitionScannerAgent.clearCache();
                }
            } catch (org.springframework.beans.factory.parsing.BeanDefinitionParsingException e) {
                LOGGER.error("Reloading XML failed: {}", e.getMessage());
            }
        }

        Set<String> beanNames = new HashSet<>();
        for (XmlBeanDefinitionScannerAgent agent : agents) {
            try {
                agent.reloadDefinition();
                beanNames.addAll(agent.beansRegistered.keySet());
            } catch (org.springframework.beans.factory.parsing.BeanDefinitionParsingException e) {
                LOGGER.error("Reloading XML failed: {}", e.getMessage());
            }
        }
        return beanNames;
    }
```

Then in 'SpringBeanReload', change it to

```java
    private void reloadXmlBeanDefinitions(boolean propertiesChanged) {
        Set<String> paths = new HashSet<>();
        if (propertiesChanged) {
            paths.addAll(beanFactoryAssistant.placeHolderXmlMapping.values());
        }
        synchronized (xmls) {
            xmls.stream().map(xml -> ResourceUtils.convertToClasspathURL(xml.getPath())).forEach(paths::add);
        }

        Set<String> result = reloadXmls(beanFactory, paths);

        synchronized (xmls) {
            xmls.clear();
        }

        processedBeans.addAll(result);
    }
```

I believe it is much more clear by doing this.